protocols/kad/src/behaviour.rs

```diff
@@ -2192,7 +2251,23 @@ pub enum KademliaEvent {
     PendingRoutablePeer {
         peer: PeerId,
         address: Multiaddr,
-    }
+    },
+
+    /// A peer sent a [`KademliaHandlerIn::PutRecord`] request and filtering is enabled.
+    ///
+    /// Cfr. [`KademliaRecordFiltering`] and [`KademliaConfig::set_record_filtering`].
+    InboundPutRecordRequest {
+        source: PeerId,
+        connection: ConnectionId,
+        record: Record,
+    },
+
+    /// A peer sent a [`KademliaHandlerIn::AddProvider`] request and filtering [`KademliaRecordFiltering::FilterBoth`] is enabled.
+    ///
+    /// Cfr. [`KademliaRecordFiltering`] and [`KademliaConfig::set_record_filtering`].
+    InboundAddProviderRequest {
+        record: ProviderRecord,
+    },
```

Comment:
How about moving these above `InboundRequestServed` as they are related to inbound requests and to routing updates?